components/Files/UploadAndList.tsx

```diff
@@ -47,6 +48,27 @@ export default function UploadAndList() {
     clearSelection();
   }, [path]);
 
+  // Handle dropped files from drag-and-drop
+  const handleFilesDropped = useCallback(
+    async (droppedFiles: File[]) => {
+      if (droppedFiles.length === 0) return;
+
+      toast.success(`Uploading ${droppedFiles.length} file${droppedFiles.length > 1 ? "s" : ""}...`);
+
+      for (const file of droppedFiles) {
+        try {
+          await handleUpload(file);
+        } catch (error) {
+          console.error(`Failed to upload ${file.name}:`, error);
+          toast.error(`Failed to upload ${file.name}`);
+        }
+      }
+
+      refreshFiles();
+    },
+    [handleUpload, refreshFiles]
+  );
```

Comment:
Open Closed Principle
- actual: UploadAndList is an existing component file being updated with new logic for drag-n-drop.
- also: UploadAndList is already over 100 lines of code and needs broken down into smaller modules.
- required: create a new lib / hook / provider for the FileUploadDragAndDrop functionality.